### PR TITLE
cmake: Link to math lib without full path

### PR DESCRIPTION
**Pull request by talregev**
Link to math lib without full path

According to @dg0yt comment:
https://github.com/curl/curl/pull/14343#issuecomment-2272712340

### CODE REVIEW

**Comment by dg0yt** [comment]
Reminder: I also wrote that this this the wrong spot for the task.

From the pkg-config files, the picure is that libm is a transitive usage requirement of WolfSSL. CURL has a Find module for WolfSSL. It is the responsibility of that module to populate `WolfSSL_LIBRARIES` with everything which needs to be linked for using WolfSSL.

File: CMakeLists.txt [hunk 1]
```diff
@@ -788,6 +788,9 @@ if(USE_NGTCP2)
       find_package(NGTCP2 REQUIRED wolfSSL)
       if(NOT WIN32)
         find_library(MATH_LIBRARY m)
+          if(MATH_LIBRARY)
+            set(MATH_LIBRARY m)
+          endif()
```
Comment (vszakats) [inline CMakeLists.txt:793]:
```suggestion
        if(MATH_LIBRARY)
          set(MATH_LIBRARY "m")
        endif()
```

**Comment by vszakats** [comment]
> Reminder: I also wrote that this this the wrong spot for the task.
> 
> From the pkg-config files, the picure is that libm is a transitive usage requirement of WolfSSL. CURL has a Find module for WolfSSL. It is the responsibility of that module to populate `WolfSSL_LIBRARIES` with everything which needs to be linked for using WolfSSL.

Agreed.

**Comment by talregev** [comment]
> > Reminder: I also wrote that this this the wrong spot for the task.
> > From the pkg-config files, the picure is that libm is a transitive usage requirement of WolfSSL. CURL has a Find module for WolfSSL. It is the responsibility of that module to populate `WolfSSL_LIBRARIES` with everything which needs to be linked for using WolfSSL.
> 
> Agreed.

@vszakats Do you still want to continue with this PR?

File: CMakeLists.txt [hunk 2]
```diff
@@ -788,6 +788,9 @@ if(USE_NGTCP2)
       find_package(NGTCP2 REQUIRED wolfSSL)
       if(NOT WIN32)
         find_library(MATH_LIBRARY m)
```
Comment (vszakats) [inline CMakeLists.txt:790]:
```suggestion
        find_library(MATH_LIBRARY "m")
```

**Comment by vszakats** [comment]
I think it improves the previous patch, but to avoid adding tech debt, I'd prefer to follow @dg0yt's suggestion.

Would you want to make an attempt to move this logic inside `CMake/FindWolfSSL.cmake`?

File: CMakeLists.txt [hunk 3]
```diff
@@ -788,6 +788,9 @@ if(USE_NGTCP2)
...
+        if(MATH_LIBRARY)
+          set(MATH_LIBRARY "m")
```
Comment (vszakats) [inline CMakeLists.txt:792]:
```suggestion
        find_library(_wolfssl_math_library "m")
        if(_wolfssl_math_library)
          set(_wolfssl_math_library "m")
```

**Comment by talregev** [comment]
I am not near computer, and I think your suggestions is important.
I don't mind that you will take over.
I will put this PR on draft.

**Comment by vszakats** [comment]
OK, I'm making an attempt in #14576.

**Comment by talregev** [comment]
> OK, I'm making an attempt in #14576.

Thank you for your hard work! and improve cmake curl, and curl itself!

**Pull request closed without merging**